### PR TITLE
DAOS-11868: Remove --with-rdma configure option from build

### PR DESCRIPTION
**Pull request by tanabarr**
Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,10 @@
+spdk (22.01.1-3) unstable; urgency=medium
+
+  [ Tom Nabarro ]
+  * Remove unnecessary --with-rdma configure option.
```
Comment (brianjmurrell) [inline debian/changelog:4]: You are not actually removing this flag in the Debian build: https://github.com/daos-stack/spdk/blob/1f940b949bca5a115411f01dd7fd25a18013ec34/debian/rules#L35

You will need to remove the line above also.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: debian/changelog (hunk 1 above)
Comment (tanabarr) [reply debian/changelog:4]: done

**Comment by tanabarr** [comment]
CI run with new packaging PR test workflow: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fspdk/detail/PR-51/4/pipeline/448
3 integration test runs (which end up running hardware small functional tests) for release/2.0, release/2.2 and master branches were triggered, release/2.0 failed, release/2.2 passed  and master failed only on known issues unrelated to the patch (https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-spdk-PR-51-master/2/tests).

Given this, what are the next steps @brianjmurrell ?

**Comment by tanabarr** [comment]
Known failures on master attributable to issues DAOS-12035 DAOS-12036.

**Review by brianjmurrell — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**